src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -55,6 +55,21 @@ ClusterResourceScheduler::ClusterResourceScheduler(
   Init(node_resources, get_used_object_store_memory, get_pull_manager_at_capacity);
 }
 
+ClusterResourceScheduler::ClusterResourceScheduler(
+    scheduling::NodeID local_node_id,
+    std::function<bool(scheduling::NodeID)> is_node_available_fn)
+    : local_node_id_(local_node_id), is_node_available_fn_(is_node_available_fn) {
+  // The GCS itself should have no resources for running tasks.
+  NodeResources local_node_resources;
+  Init(local_node_resources,
+       /*get_used_object_store_memory=*/nullptr,
+       /*get_pull_manager_at_capacity=*/nullptr);
+  // Add the Nil local node to cluster_resource_manager_, just in order to accommodate the
+  // hybrid scheduling policy. This local node would not be selected because
+  // `is_node_available_fn_` excludes it.
+  cluster_resource_manager_->AddOrUpdateNode(local_node_id_, local_node_resources);
```

Comment:
You can random a node id as the `local_node_id_` when `ClusterResourceScheduler` is used in gcs scheduling, then reuse `ClusterResourceScheduler(scheduling::NodeID local_node_id, const NodeResources &local_node_resources, std::function<bool(scheduling::NodeID)> is_node_available_fn)` constructor.

Reply:
BTW, remove the default constructor of `ClusterResourceScheduler` and add RAY_CHECK(!local_node_id_.IsNil()) inside `Init(...)` method.